beacon-chain/core/altair/sync_committee.go

```diff
@@ -0,0 +1,163 @@
+//        shuffled_index = compute_shuffled_index(uint64(i % active_validator_count), active_validator_count, seed)
+//        candidate_index = active_validator_indices[shuffled_index]
+//        random_byte = hash(seed + uint_to_bytes(uint64(i // 32)))[i % 32]
+//        effective_balance = state.validators[candidate_index].effective_balance
+//        if effective_balance * MAX_RANDOM_BYTE >= MAX_EFFECTIVE_BALANCE * random_byte:
+//            sync_committee_indices.append(candidate_index)
+//        i += 1
+//    return sync_committee_indices
+func NextSyncCommitteeIndices(state state.BeaconStateAltair) ([]types.ValidatorIndex, error) {
+	epoch := helpers.NextEpoch(state)
+	indices, err := helpers.ActiveValidatorIndices(state, epoch)
+	if err != nil {
+		return nil, err
+	}
+	count := uint64(len(indices))
+	seed, err := helpers.Seed(state, epoch, params.BeaconConfig().DomainSyncCommittee)
+	if err != nil {
+		return nil, err
+	}
+	i := types.ValidatorIndex(0)
+	cIndices := make([]types.ValidatorIndex, 0, params.BeaconConfig().SyncCommitteeSize)
+	hashFunc := hashutil.CustomSHA256Hasher()
+	maxRandomByte := uint64(1<<8 - 1)
+	for uint64(len(cIndices)) < params.BeaconConfig().SyncCommitteeSize {
+		sIndex, err := helpers.ComputeShuffledIndex(i.Mod(count), count, seed, true)
+		if err != nil {
+			return nil, err
+		}
+
+		b := append(seed[:], bytesutil.Bytes8(uint64(i.Div(32)))...)
+		randomByte := hashFunc(b)[i%32]
+		cIndex := indices[sIndex]
+		v, err := state.ValidatorAtIndexReadOnly(cIndex)
+		if err != nil {
+			return nil, err
+		}
+
+		effectiveBal := v.EffectiveBalance()
+		if effectiveBal*maxRandomByte >= params.BeaconConfig().MaxEffectiveBalance*uint64(randomByte) {
+			cIndices = append(cIndices, cIndex)
+		}
+		i++
+	}
+	return cIndices, nil
+}
+
+// SyncSubCommitteePubkeys returns the pubkeys participating in a sync subcommittee.
+//
+// def get_sync_subcommittee_pubkeys(state: BeaconState, subcommittee_index: uint64) -> Sequence[BLSPubkey]:
+//    # Committees assigned to `slot` sign for `slot - 1`
+//    # This creates the exceptional logic below when transitioning between sync committee periods
+//    next_slot_epoch = compute_epoch_at_slot(Slot(state.slot + 1))
+//    if compute_sync_committee_period(get_current_epoch(state)) == compute_sync_committee_period(next_slot_epoch):
+//        sync_committee = state.current_sync_committee
+//    else:
+//        sync_committee = state.next_sync_committee
+//
+//    # Return pubkeys for the subcommittee index
+//    sync_subcommittee_size = SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT
+//    i = subcommittee_index * sync_subcommittee_size
+//    return sync_committee.pubkeys[i:i + sync_subcommittee_size]
+func SyncSubCommitteePubkeys(syncCommittee *ethpb.SyncCommittee, subComIdx types.CommitteeIndex) ([][]byte, error) {
+	subCommSize := params.BeaconConfig().SyncCommitteeSize / params.BeaconConfig().SyncCommitteeSubnetCount
```

Comment:
```suggestion
	cfg := params.BeaconConfig()
	subCommSize := cfg.SyncCommitteeSize / cfg.SyncCommitteeSubnetCount
```

Not sure if this helps in performance or just code clarity / readability.